packages/validator/test/spec/spec.test.ts

```diff
@@ -22,42 +22,44 @@ describe("slashing-protection-interchange-tests", () => {
   let db: LevelDbController;
   let slashingProtection: SlashingProtection;
 
-  before(async () => {
+  beforeAll(async () => {
     db = await LevelDbController.create({name: dbLocation}, {logger: testLogger()});
     slashingProtection = new SlashingProtection(db);
   });
 
-  after(async () => {
+  afterAll(async () => {
     await db.close();
     rimraf.sync(dbLocation);
   });
 
   for (const testCase of testCases) {
     describe(testCase.name, () => {
       for (const step of testCase.steps) {
-        beforeEach(async () => {
-          await db.clear();
-        });
+        // If there is no `it` block then we should skip to avoid running `beforeEach` hooks
+        if (step.blocks.length === 0 && step.attestations.length === 0) {
+          continue;
+        }
```

Comment:
This is a very interesting case to show difference of both test runners. 

1. Mocha don't run the `beforeEach` which is defined within the loop without `it`.
2. Vitest consider all `it` blocks and `beforeEach` blocks as children of the `describe` and run two `beforeEach` for single `it` block.